bin/node-sass

```diff
@@ -162,8 +207,7 @@ function getOptions(args, options) {
  */
 
 function watch(options, emitter) {
-  var glob = options.recursive ? '**/*.{sass,scss}' : '*.{sass,scss}';
-  var src = isSassFile(options.src) ? options.src : path.join(options.src, glob);
+  var src = isSassFile(options.src) ? options.src : path.join(options.src, globPattern(options));
```

Comment:
I might be wrong here, but doesn't your `options.directory` solve the same problem as this `isSassFile` call?  Maybe you can just use that here, and replace the second `options.src` with `options.directory`.  Also you can replace `path.resolve(path.dirname(src))` on the line below if we already know we have a directory.